### PR TITLE
ci(github): bound clean-state cleanup

### PR DESCRIPTION
**Pull request by jamesryancooper**
Policy reference: `.octon/framework/execution-roles/practices/pull-request-standards.md`
Reminder: Run `@codex review`.

## What

Bounds PR Clean State Enforcer closed-head-ref cleanup and treats GitHub API rate limits during cleanup or health issue sync as deferred cleanup instead of hard open-PR clean-state failure.

Also updates the autonomy label validator message from legacy agency wording to execution-role terminology.

## Why

After #362 merged, the post-merge clean-state run exhausted the GitHub installation API rate limit while iterating old closed PR head refs. That failure was cleanup-path drift, not a dependency or Octon runtime failure.

No-Issue: follow-up to #362 post-merge clean-state failure.

## How

The workflow now limits closed PR head-ref deletion to `AUTONOMY_CLOSED_HEAD_CLEANUP_LIMIT` per run, fetches only one bounded closed-PR page, stops cleanup on rate-limit responses, and records cleanup deferral in the job summary.

The open-PR attention path remains the canonical clean-state signal. Rate-limited closed-ref cleanup is treated as deferred housekeeping because it does not change whether open PRs require action.

## Profile Selection Receipt

- Semantic version source(s): `Cargo.toml` release state remains pre-1.0.
- Release state (`pre-1.0` or `stable`): `pre-1.0`.
- `change_profile` (`atomic` or `transitional`): `atomic`.
- Hard-gate facts (downtime, coordination, migration/backfill, rollback, blast radius, compliance): no downtime, no migration/backfill, workflow-only blast radius, no compliance data expansion.
- Tie-break status: no tie-break required.
- `transitional_exception_note` (required when `pre-1.0` + `transitional`): n/a.

## Implementation Plan

- Workstreams and scope: update clean-state workflow cleanup behavior, update execution-role runbook, fix stale validator wording.
- Public interfaces/types/contracts affected: GitHub Actions workflow behavior and documented autonomy runbook behavior.
- Test scenarios: YAML parse, embedded JavaScript syntax check, autonomy label validation, unified execution closure assertion, diff hygiene.
- Assumptions/defaults: default closed head cleanup limit is 25; maintainers may tune `AUTONOMY_CLOSED_HEAD_CLEANUP_LIMIT`.

## Impact Map (code, tests, docs, contracts)

- Code/workflow: `.github/workflows/pr-clean-state-enforcer.yml`.
- Validation script: `.octon/framework/execution-roles/_ops/scripts/validate/validate-autonomy-labels.sh`.
- Docs: `.octon/framework/execution-roles/practices/github-autonomy-runbook.md`.
- Contracts: no schema or runtime contract changes.

## Compliance Receipt

- [x] Selected exactly one execution profile before planning/implementation.
- [x] Applied release-maturity gate from semantic version.
- [x] Enforced pre-1.0 default (`atomic`) unless hard gates required `transitional`.
- [x] Included `transitional_exception_note` when required.
- [x] Included tie-break escalation when applicable.
- [x] Updated impacted contracts/docs/tests.
- [x] Honored charter change-control constraint (no direct principles charter edits without override evidence).

## Exceptions/Escalations

none

## Tradeoffs

Closed-ref cleanup may take multiple workflow runs when there are many stale branches. That is preferable to exhausting the GitHub installation API quota and causing unrelated checks to fail.

## Testing

- `yq e . .github/workflows/pr-clean-state-enforcer.yml >/dev/null`
- `yq -r .jobs["enforce-clean-state"].steps[0].with.script .github/workflows/pr-clean-state-enforcer.yml | node --input-type=module --check -`
- `bash .octon/framework/execution-roles/_ops/scripts/validate/validate-autonomy-labels.sh`
- `bash .octon/framework/assurance/governance/_ops/scripts/assert-unified-execution-closure.sh`
- `git diff --cached --check`

## Rollout

Merge workflow change through the normal PR lane. The next clean-state run uses the bounded cleanup behavior.

## Convivial Purpose Check

- [x] Feature expands genuine user capability (not synthetic engagement).
- [x] Attention and interruption behavior are justified and user-controllable.
- [x] No manipulative patterns or dark-pattern mechanics are introduced.
- [x] Data collection/extraction risk is minimal and explicitly justified.

## Risk Rubric

- Risk class: [ ] Trivial [ ] Low [ ] Medium [x] High
- Rollback plan: revert this workflow/docs/script commit.
- Rollback handle: PR #371 / commit `e5e3a3df`.
- Flags changed (name, owner, expiry, rollout): `AUTONOMY_CLOSED_HEAD_CLEANUP_LIMIT`, repo variable optional, default 25, no expiry.
- Autonomy eligibility: [ ] autonomy:auto-merge [x] autonomy:no-automerge

## Contracts and Threat Model

- OpenAPI/JSON-Schema changes: none.
- Threat model update/link: GitHub API quota exhaustion is reduced by bounding cleanup and failing soft on cleanup-only rate limits.

## Observability and Performance

- Traces/logs/metrics for changed flows: job summary now reports deleted/skipped closed refs, cleanup limit, cleanup rate limit status, and health issue sync skip status.
- Representative traces for high risk changes: CI check logs for PR #371.
- Performance or bundle impact: fewer GitHub API calls per clean-state run.

## License and Provenance

- New dependencies and licenses: none.
- Generated code/templates provenance notes: none.

## Checklist

- [x] Requirements met; edge cases handled
- [x] Security reviewed (authz, input validation, secrets)
- [x] Tests added or updated
- [x] Observability updated (logs, metrics, traces) if needed
- [x] No speculative abstractions or unnecessary complexity
- [x] Conventions followed; no drift introduced
- [x] Non-obvious decisions documented (comments, ADR)
- [x] Reviewer feedback addressed with fix, commit, push, reply; reviewer-owned threads left for reviewer or maintainer confirmation

## Screenshots / Notes

This PR exists because #362 merged successfully but exposed a post-merge clean-state rate-limit failure in the existing workflow.

### CODE REVIEW

File: .github/workflows/pr-clean-state-enforcer.yml [hunk 1]
```diff
@@ -41,11 +42,20 @@ jobs:
...
+            );
 
             const attentionLabel = "autonomy:attention-required";
             const healthLabel = "ops:autonomy-health";
             const attentionIssueTitle = "[autonomy-health] open-pr attention required";
 
+            function isRateLimit(error) {
+              const message = String(error?.message || error?.response?.data?.message || "");
+              return error?.status === 403 && /rate limit/i.test(message);
+            }
```
[the remark this answers is not in the document]
Comment (jamesryancooper) [reply .github/workflows/pr-clean-state-enforcer.yml:57]: Fixed in 505c68e36: `isRateLimit` now treats HTTP 429 responses as rate-limit deferrals as well as 403 rate-limit messages, so secondary rate limits fail soft in the same paths.
